src/responder/multiResponder.ts

```diff
@@ -63,16 +64,16 @@ export class MultiResponder {
                         `Cannot add to queue. Max queue depth ${this.zStore.queue.maxQueueDepth} reached.`
                     );
                 }
-
+                
                 // form a queue item request
                 const txIdentifier = new PisaTransactionIdentifier(
                     this.chainId,
-                    appointment.data,
-                    appointment.contractAddress,
+                    appointment.encodeForResponse(),
+                    this.pisaContractAddress,
                     new BigNumber(0),
-                    appointment.gasLimit
+                    // it appears that sometimes pisa requires a lot of gas to function - resetting data shards?
+                    appointment.gasLimit.add(400000)
```

Comment:
Seems a bit hackish, perhaps we should have an open issue to keep track.

Reply:
Yeah, I agree. #292